remoteworkitem/trackeritem_repository.go

```diff
@@ -29,77 +32,144 @@ func upload(db *gorm.DB, tID int, item TrackerItemContent) error {
+	}
+
+	return upsert(db, *workItem)
+}
+
+func bindAssignees(db *gorm.DB, remoteWorkItem RemoteWorkItem, providerType string) (*app.WorkItem, error) {
+	identityRepository := account.NewIdentityRepository(db)
+	workItem := app.WorkItem{
+		ID:     remoteWorkItem.ID,
+		Type:   remoteWorkItem.Type,
+		Fields: make(map[string]interface{}),
+	}
+	// copy all fields from remoteworkitem into result workitem
+	for fieldName, fieldValue := range remoteWorkItem.Fields {
+		if fieldName == remoteAssigneeProfileURLs {
+			if fieldValue == nil {
+				workItem.Fields[workitem.SystemAssignees] = make([]string, 0)
+				continue
+			}
+			identities := make([]string, 0)
+			assigneeLogins := fieldValue.([]string)
+			assigneeProfileURLs := remoteWorkItem.Fields[remoteAssigneeProfileURLs].([]string)
+			for i, assigneeLogin := range assigneeLogins {
+				assigneeProfileURL := assigneeProfileURLs[i]
+				log.Debug(nil, map[string]interface{}{
+					"pkg":      "remoteworkitem",
+					"workitem": workItem,
+				}, "Looking for identity of user with profile URL=%s\n", assigneeProfileURL)
+				// bind the assignee to an existing identity, or create a new one
+				identity, err := identityRepository.First(account.IdentityFilterByProfileURL(assigneeProfileURL))
+				if err != nil {
+					return nil, err
+				}
+				if identity == nil {
+					// create the identity if it does not exist yet
+					log.Debug(nil, map[string]interface{}{
+						"pkg":      "remoteworkitem",
+						"workitem": workItem,
+					}, "Creating an identity for username '%s' with profile '%s' on '%s'\n", assigneeLogin, assigneeProfileURL, providerType)
+					identity := &account.Identity{
+						ProviderType: providerType,
+						Username:     assigneeLogin,
+						ProfileURL:   assigneeProfileURL,
+					}
+					err = identityRepository.Create(context.Background(), identity)
+					if err != nil {
+						return nil, err
+					}
+					identities = append(identities, identity.ID.String())
+				} else {
+					// use existing identity
+					log.Debug(nil, map[string]interface{}{
+						"pkg":      "remoteworkitem",
+						"workitem": workItem,
+					}, "Using existing identity with ID: %v", identity.ID.String())
+					identities = append(identities, identity.ID.String())
+				}
+			}
+			// associate the identities to the work item
+			workItem.Fields[workitem.SystemAssignees] = identities
+		} else if fieldName == remoteAssigneeProfileURLs {
+			// ignore here, it is being processed above
+		} else {
+			// copy other fields
+			workItem.Fields[fieldName] = fieldValue
+		}
 	}
+	return &workItem, nil
+}
 
+func upsert(db *gorm.DB, workItem app.WorkItem) (*app.WorkItem, error) {
+	wir := workitem.NewWorkItemRepository(db)
 	// Get the remote item identifier ( which is currently the url ) to check if the work item exists in the database.
 	workItemRemoteID := workItem.Fields[workitem.SystemRemoteItemID]
-
+	log.Info(nil, map[string]interface{}{
+		"pkg":      "remoteworkitem",
+		"workitem": workItem,
```

Comment:
In order to track the activity, we usually use the workitem ID e.g `"wiID": workItem.ID.String()`. You can also log the workitem itself but we prefer to follow the same pattern. To then use Elasticsearch or any other tool to analysis the logs. @xcoulon Could you change that in your logs ?